integrations/opsgenie/client.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, AsyncGenerator, Optional
+import httpx
+from loguru import logger
+from utils import ObjectKind, RESOURCE_API_VERSIONS
+from port_ocean.context.event import event
+
+
+PAGE_SIZE = 50
+
+
+class OpsGenieClient:
+    def __init__(self, token: str, api_url: str):
+        self.token = token
+        self.api_url = api_url
+        self.http_client = httpx.AsyncClient(headers=self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {"Authorization": f"GenieKey {self.token}"}
+
+    async def get_resource_api_version(self, resource_type: ObjectKind) -> str:
+        return RESOURCE_API_VERSIONS.get(resource_type, "v2")
+
+    async def _get_single_resource(
+        self,
+        url: str,
+        query_params: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        try:
+            response = await self.http_client.get(url=url, params=query_params)
+            response.raise_for_status()
+            return response.json()
+
+        except httpx.HTTPStatusError as e:
+            logger.error(
+                f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+            )
+            raise
+
+    async def get_paginated_resources(
+        self, resource_type: ObjectKind
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        if cache := event.attributes.get(resource_type.value):
+            yield cache
+            return
+        api_version = await self.get_resource_api_version(resource_type)
+        url = f"{self.api_url}/{api_version}/{resource_type.value}s"
+        pagination_params: dict[str, Any] = {"limit": PAGE_SIZE}
+
+        while url:
+            try:
+                response = await self._get_single_resource(
+                    url=url, query_params=pagination_params
+                )
+                yield response["data"]
+
+                url = response.get("paging", {}).get("next")
+            except httpx.HTTPStatusError as e:
+                logger.error(
+                    f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+                )
+                raise
+
+    async def get_alert(self, identifier: str) -> dict[str, Any]:
+        api_version = await self.get_resource_api_version(ObjectKind.ALERT)
+        url = f"{self.api_url}/{api_version}/alerts/{identifier}"
+        alert_data = (await self._get_single_resource(url))["data"]
+        return await self.get_related_incident_by_alert(alert_data)
+
+    async def get_oncall_team(self, identifier: str) -> dict[str, Any]:
+        cache_key = f"{ObjectKind.TEAM}-{identifier}"
+        if cache := event.attributes.get(cache_key):
+            return cache
```

Comment:
sometimes you are caching it in the function sometimes u r doing it somewhere else
align all caches to work and look the same